Archives/ClmFile.cpp

```diff
@@ -180,15 +135,15 @@ namespace Archives
 	{
 		headerOut.riffHeader.riffTag = RIFF;
 		headerOut.riffHeader.waveTag = WAVE;
-		headerOut.riffHeader.chunkSize = sizeof(headerOut.formatChunk) + 12 + m_IndexEntry[fileIndex].dataLength;
+		headerOut.riffHeader.chunkSize = sizeof(headerOut.riffHeader.waveTag) + sizeof(FormatChunk) + sizeof(ChunkHeader) + indexEntries[fileIndex].dataLength;
 
 		headerOut.formatChunk.fmtTag = FMT;
-		headerOut.formatChunk.formatSize = sizeof(headerOut.formatChunk.waveFormat);
-		headerOut.formatChunk.waveFormat = m_WaveFormat;
+		headerOut.formatChunk.formatSize = sizeof(WaveFormatEx);
```

Comment:
It may be better to use the size of the field, rather than the size of the struct. If the field declaration is changed, the sizes will auto update to match the declaration. (Perhaps someday there will be a WaveFormatExEx?)

Reply:
Okay, I'll change it back. I was trying to make it more readable but I see where you are coming from. Lets just not plan on making a WaveFormatExEx if we can help it!